docs/services/endlessh.md

```diff
@@ -0,0 +1,70 @@
+# Endlessh
+
+[Endlessh-go](https://github.com/shizunge/endlessh-go) is a A golang implementation of endlessh, a ssh trapit. Installing it is powered by the [mother-of-all-self-hosting/ansible-role-endlessh](https://github.com/mother-of-all-self-hosting/ansible-role-endlessh) Ansible role.
+
+## Dependencies
+
+This service requires the following other services:
+
+- (optionally) [Traefik](traefik.md) - a reverse-proxy server for exposing endlessh publicly
+- (optionally) [Prometheus](./prometheus.md) - a database for storing metrics
+- (optionally) [Grafana](./grafana.md) - a web UI that can query the prometheus datasource (connection) and display the logs
+
+## Installing
+
+To configure and install endlessh on your own server(s), you should use a playbook like [Mother of all self-hosting](https://github.com/mother-of-all-self-hosting/mash-playbook) or write your own.
+
+## Configuration
+
+To enable this service, add the following configuration to your `vars.yml` file and re-run the [installation](../installing.md) process:
+
+```yaml
+########################################################################
+#                                                                      #
+# endlessh                                                                 #
+#                                                                      #
+########################################################################
+
+endlessh_enabled: true
+
+########################################################################
+#                                                                      #
+# /endlessh                                                                #
+#                                                                      #
+########################################################################
+```
+See the full list of options in the [default/main.yml](default/main.yml) file
+
+## Exporting Prometheus metrics
+
+Default port is 2112
+
+Default entrypoint is /metrics
+
+use environment variable 
+```yaml
+endlessh_container_extra_arguments_custom:
+  - "-enable_prometheus"
+```
+
+### Exposing the web interface
+
+By setting a hostname and optionally a path prefix, you can expose endlessh publicly.
```

Comment:
Does not setting up a hostname disable exposure? Or will it complain that it needs it?

Looking at the code, it seems like it's completely optional. We should make this more explicit.

-----

Will the web UI show useful stuff even without Prometheus and Grafana? I suppose it would.

Reply:
Endlessh-go has no Web UI, `endlessh_hostname` is only used for metrics exposure.
Therefor, setting a Hostname is only to required to expose metrics publicly.
I Fixed the documentation.